tokio-console/src/view/styles.rs

```diff
@@ -126,39 +156,104 @@ impl Styles {
+                FormattedDuration::Debug(s) if s.ends_with("µs") || s.ends_with("us") => {
+                    fg_style(Color::Magenta)
+                }
+                FormattedDuration::Debug(s) if s.ends_with("ms") => fg_style(Color::Red),
+                FormattedDuration::Debug(s) if s.ends_with('s') => fg_style(Color::Yellow),
                 _ => Style::default(),
             },
-            Palette::Ansi256 | Palette::All => match text.as_ref() {
-                s if s.ends_with("ps") => fg_style(Color::Indexed(40)), // green 3
-                s if s.ends_with("ns") => fg_style(Color::Indexed(41)), // spring green 3
-                s if s.ends_with("µs") || s.ends_with("us") => fg_style(Color::Indexed(42)), // spring green 2
-                s if s.ends_with("ms") => fg_style(Color::Indexed(43)), // cyan 3
-                s if s.ends_with('s') => fg_style(Color::Indexed(44)),  // dark turquoise,
+            Palette::Ansi256 | Palette::All => match &formatted {
+                FormattedDuration::Days(_) => fg_style(Color::Indexed(33)), // dodger blue 1
+                FormattedDuration::DaysHours(_) => fg_style(Color::Indexed(33)), // dodger blue 1
+                FormattedDuration::HoursMinutes(_) => fg_style(Color::Indexed(39)), // deep sky blue 1
+                FormattedDuration::MinutesSeconds(_) => fg_style(Color::Indexed(45)), // turquoise 2
+                FormattedDuration::Debug(s) if s.ends_with("ps") => fg_style(Color::Indexed(40)), // green 3
+                FormattedDuration::Debug(s) if s.ends_with("ns") => fg_style(Color::Indexed(41)), // spring green 3
+                FormattedDuration::Debug(s) if s.ends_with("µs") || s.ends_with("us") => {
+                    fg_style(Color::Indexed(42))
+                } // spring green 2
+                FormattedDuration::Debug(s) if s.ends_with("ms") => fg_style(Color::Indexed(43)), // cyan 3
+                FormattedDuration::Debug(s) if s.ends_with('s') => fg_style(Color::Indexed(44)), // dark turquoise,
                 _ => Style::default(),
             },
         };
 
-        Span::styled(text, style)
+        Span::styled(formatted.into_inner(), style)
+    }
+
+    fn duration_text(&self, dur: Duration, width: usize, prec: usize) -> FormattedDuration {
+        let secs = dur.as_secs();
+
+        if secs >= 60 * 60 * 24 * 100 {
+            let days = secs / (60 * 60 * 24);
+            FormattedDuration::Days(format!("{days:>width$}d", days = days, width = width))
+        } else if secs >= 60 * 60 * 24 {
+            let hours = secs / (60 * 60);
+            FormattedDuration::DaysHours(format!(
+                "{days:>leading_width$}d{hours:02.0}h",
+                days = hours / 24,
+                hours = hours % 24,
+                // Subtract the known 4 characters that trail the days value.
+                leading_width = if width > 4 { width - 4 } else { 0 }
```

Comment:
nit, take it or leave it: this looks like a manual implementation of `saturating_sub`:
```suggestion
                leading_width = width.saturating_sub(4),
```

Reply:
Nice, thanks for the tip!